### PR TITLE
Tiled rendering

### PR DESCRIPTION
**Pull request by guidow**

### CODE REVIEW

**Comment by opalmer** [comment]
@guidow, you ready for a review of this?

**Comment by guidow** [comment]
Yes, this is ready for review now.

File: pyfarm/master/api/jobs.py [hunk 1]
```diff
@@ -443,13 +443,28 @@ def post(self):
...
+
         current_frame = start
         while current_frame <= end:
-            task = Task()
-            task.job = job
-            task.frame = current_frame
-            task.priority = job.priority
-            db.session.add(task)
+            if num_tiles:
+                for tile in range(0, num_tiles - 1):
```
Comment (opalmer) [inline pyfarm/master/api/jobs.py]: Use the appropriate range:

```python
try:
  range_ = xrange
except NameError:
  range_ = range
```

File: pyfarm/master/api/jobs.py [hunk 2]
```diff
@@ -791,11 +810,20 @@ def post(self, job_name):
                     frames_to_create.remove(task.frame)
 
             for frame in frames_to_create:
-                task = Task()
-                task.job = job
-                task.frame = frame
-                task.priority = job.priority
-                db.session.add(task)
+                if job.num_tiles:
+                    for tile in range(0, job.num_tiles - 1):
```
Comment (opalmer) [inline pyfarm/master/api/jobs.py]: Use the appropriate range so we get a generator:

```python
try:
  range_ = xrange
except NameError:
  range_ = range
```

File: pyfarm/models/job.py [hunk 3]
```diff
@@ -548,11 +553,20 @@ def alter_frame_range(self, start, end, by):
                 frames_to_create.remove(task.frame)
 
         for frame in frames_to_create:
-            task = Task()
-            task.job = self
-            task.frame = frame
-            task.priority = self.priority
-            db.session.add(task)
+            if self.num_tiles:
+                for tile in range(0, self.num_tiles - 1):
```
Comment (opalmer) [inline pyfarm/models/job.py]: Use the appropriate range so we get a generator:

```python
try:
  range_ = xrange
except NameError:
  range_ = range
```

**Comment by opalmer** [comment]
Seems generally ok and much simpler than I guess I was expecting.  I thought it might be beneficial to offer two modes of tiling, one where the jobtype on the agent determines how to tile the image and another where the user can.  But...this setup will probably work fine and it's not like we couldn't include something more customizable later.

**Comment by guidow** [comment]
I've rebase this branch to master to make it mergable again.

**Comment by guidow** [comment]
Any more objections to mergeing this?

**Comment by opalmer** [comment]
Nope none, sorry for the delay got caught up on another project.  Merge when ready.

**Pull request merged**